core/src/main/java/cucumber/runner/PickleStepTestStep.java

```diff
@@ -39,6 +40,10 @@ boolean run(TestCase testCase, EventBus bus, Scenario scenario, boolean skipStep
         for (HookTestStep before : beforeStepHookSteps) {
             skipNextStep |= before.run(testCase, bus,  scenario, skipSteps);
         }
+        
```

Comment:
We should avoid using singletons and setters in this codebase and use constructor parameters instead.